platform/darwin/src/NSExpression+MGLAdditions.h

```diff
@@ -42,6 +90,133 @@ NS_ASSUME_NONNULL_BEGIN
  */
 @property (nonatomic, readonly) id mgl_jsonExpressionObject;
 
+/**
+ Returns a constant expression containing an `NSString`.
+ 
+ This is equivalent to call `[NSExpression expressionForConstantValue:]`.
+ 
+ @param string The string constant.
+ */
++ (instancetype)mgl_expressionForString:(nonnull NSString *)string;
```

Comment:
Regarding https://github.com/mapbox/mapbox-gl-native/pull/11278#discussion_r172640159, I think we should remove these initializers.

The original goal in #8074 was to reuse standard NSExpression APIs as much as possible, so that developers would lean on non-Mapbox resources and experiences to craft their runtime styling code. Someone who already isn’t sure whether to use `NSExpression(format:)` or `NSExpression(forConstantValue:)` now has to contend with `NSExpression.mgl_expression(for:)` as well. If they’re used to NSExpression in other contexts, they may wonder what `NSExpression.mgl_expression(for:)` provides that the other methods do not.

#11016 is meant to ease the transition for developers accustomed to the more strongly-typed MGLStyleValue API, but not necessarily to make NSExpression more strongly typed. In fact, these initializers add a false sense of type safety. The MGLStyleLayer properties no longer make use of lightweight generic types, so the following code can still crash the application at runtime:

```swift
symbolStyleLayer.text = NSExpression.mgl_expression(for: number)
```

Finally, the approach here is inconsistent with MGLStyleValue. NSExpression’s built-in initializers distinguish between constant values and functions, just like MGLStyleValue’s initializers. By contrast, `NSExpression.mgl_expression(for:)` emphasizes the data type instead of the constant nature of the value.

Reply:
I agree with you. These are repetitive. Although these methods attempt to avoid the confusion of having to know the difference or when to use a specific method. For someone new to expressions will they choose `expressionForConstantValue` or `expressionWithFormat` for a string/color/value? Also does `expressionForConstantValue` means I won't be able to modify it later? These sort of questions are what these methods try to make easier to reason about.

Reply:
> or someone new to expressions will they choose expressionForConstantValue or expressionWithFormat for a string/color/value?

I think it would be the same as anyone newly discovering NSExpression or NSPredicate in general. There have always been multiple ways of creating one, but the format string is preferable unless you have an overriding need for type safety. Among Objective-C developers, that would be [pretty rare](http://nshipster.com/nsexpression/). I originally brought up the idea of providing these initializers for Swift developers who are migrating from style functions, which are strongly typed to a fault.

> Also does expressionForConstantValue means I won't be able to modify it later?

“Constant” here means the evaluated value doesn’t change based on the context; no expression is mutable.

Reply:
Agree. This PR no longer has value based initializers and focuses on the generic types.